plugins/basestate/BaseStateDriver.cpp

```diff
@@ -94,6 +94,8 @@ class yarp::dev::gzyarp::BaseStateDriver : public yarp::dev::DeviceDriver,
 
     yarp::os::Stamp getLastInputStamp()
     {
+        std::lock_guard<std::mutex> lock(m_baseLinkData->m_mutex);
+        m_lastTimestamp.update(m_baseLinkData->m_simTime);
```

Comment:
The `yarp::os::Stamp` also have a sequence number, that is increased for any call to update (see https://yarp.it/latest/classyarp_1_1os_1_1Stamp.html#a2b836a2044d0929ad4e5a339ae0c1635). It may be strange that the sequence number tracks the number of times `getLastInputStamp()` has been called, probably it could make sense to move this `m_lastTimestamp.update` call to when `m_baseStateData.m_simTime` is updated?

Reply:
Updated in b0c3637958f9a6d029215c5aea4b1ae436763270